src/main/webapp/app/home/home.component.ts

```diff
@@ -150,7 +150,9 @@ export class HomeComponent implements OnInit, AfterViewInit {
     currentUserCallback(account: User) {
         this.account = account;
         if (account) {
-            this.router.navigate(['courses']);
+            // TODO: Remove redirect after summer 2021 term. New deep links should no longer ues /#.
+            const url = this.router.url.startsWith('/#') ? this.router.url.substr(2) : 'courses';
```

Comment:
I assume `this.router.url` is either `#/courses` (old) or just `courses` (new). `substr` would trim `#/courses` down to just `courses`, which means `url` will always be `'courses'`, so I'm not sure why this is needed. (In fact this looks like it should be `#/` instead of `/#` meaning this already always takes the else branch with just `'courses'`? Except if the input is something like `/#/courses` in which case the order doesn't matter.)

Reply:
`this.router.url` would be `/#/course-management` etc. This is the case because the Fragment is now ignored in that senses and the empty route is routed to the HomeComponent.

Reply:
Ah, thanks for the clarification.